### PR TITLE
[ISSUE-647] Upload artifacts to GH

### LINKED ISSUE #647: Attach new e2e test logs to GitHub action artifacts
Make artifacts available in GH workflow

### PR DESCRIPTION
**Pull request by rkiselenko**
### Resolves #647

Add junit report, logs to artifacts

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

How it looks https://github.com/rkiselenko/csi-baremetal/runs/4518289541

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -159,11 +159,17 @@ jobs:
     - name: Make test
       run: make test-short-ci CSI_VERSION=$CSI_VERSION OPERATOR_VERSION=$CSI_OPERATOR_VERSION CHARTS_DIR=${{env.CSI_BAREMETAL_OPERATOR_DIR}}/charts
 
+    - name: Publish Unit Test Results
+      uses: EnricoMi/publish-unit-test-result-action@v1
+      if: always()
+      with:
+        files: test/e2e/reports/report.xml
+
     - name: Upload report to artifacts
```
Comment (safronovD) [inline .github/workflows/e2e.yml:168]: Why do you set `if: always()` for the first upload only?

**Review by safronovD — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -159,11 +159,17 @@ jobs:
     - name: Make test
       run: make test-short-ci CSI_VERSION=$CSI_VERSION OPERATOR_VERSION=$CSI_OPERATOR_VERSION CHARTS_DIR=${{env.CSI_BAREMETAL_OPERATOR_DIR}}/charts
 
+    - name: Publish Unit Test Results
+      uses: EnricoMi/publish-unit-test-result-action@v1
+      if: always()
```
Comment (mishoyama) [inline .github/workflows/e2e.yml:164]: Do we need to check presence of report before uploading it?

Comment (rkiselenko) [reply .github/workflows/e2e.yml]: The if: always() clause guarantees that this action always runs, even if earlier steps (e.g., the unit test step) in your workflow fail.

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -159,17 +159,19 @@ jobs:
...
+      uses: EnricoMi/publish-unit-test-result-action@v1
       with:
-        name: report.xml
-        path: test/e2e/report.xml
-    
-    - name: Upload log to artifacts
+        files: test/e2e/reports/report.xml
+
+    - name: Tar files
+      run: tar -cvf reports.tar test/e2e/reports/
```
Comment (mishoyama) [inline .github/workflows/e2e.yml:168]: Can we use `zip` since `tar` is not available on Windows?

**Review by mishoyama — APPROVED** [review]

**Pull request merged**